c2corg_api/views/document_tag.py

```diff
@@ -0,0 +1,152 @@
+import logging
+
+from c2corg_api import DBSession
+from c2corg_api.models.document import Document
+from c2corg_api.models.document_tag import DocumentTag, DocumentTagLog
+from c2corg_api.models.route import ROUTE_TYPE
+from c2corg_api.search.notify_sync import notify_es_syncer
+from c2corg_api.views import cors_policy, restricted_json_view
+from c2corg_api.views.validation import create_int_validator
+from colander import MappingSchema, SchemaNode, Integer, required
+from cornice.resource import resource
+from cornice.validators import colander_body_validator
+from pyramid.httpexceptions import HTTPBadRequest
+
+log = logging.getLogger(__name__)
+
+
+class DocumentTagSchema(MappingSchema):
+    document_id = SchemaNode(Integer(), missing=required)
+
+
+def get_tag_relation(user_id, document_id):
+    return DBSession. \
+        query(DocumentTag). \
+        filter(DocumentTag.user_id == user_id). \
+        filter(DocumentTag.document_id == document_id). \
+        first()
+
+validate_document_id = create_int_validator('document_id')
+
+
+def validate_document(request, **kwargs):
+    """ Check that the document exists.
```

Comment:
Perhaps check also that the document type is actually of type ``ROUTE_TYPE`` (or whatever type passed if other types are supported later).